cms/djangoapps/contentstore/views/course.py

```diff
@@ -1082,19 +1056,30 @@ def _modify_tabs_to_components(request, course_module):
             # user has failed to put iterable value into advanced component list.
             # return immediately and let validation handle.
             return
+        update_tab(course_tabs, tab_type, tab_enabled)
 
-        if tab_enabled:
-            # check passed, some of this component_types are present, adding tab
-            if _add_tab(request, tab_type, course_module):
-                # tab indeed was added, the change needs to propagate
-                tabs_changed = True
-        else:
-            # the tab should not be present (anymore)
-            if _remove_tab(request, tab_type, course_module):
-                # tab indeed was removed, the change needs to propagate
-                tabs_changed = True
+    # Additionally update any persistent tabs provided by course views
+    for tab_type in CourseTabManager.get_tab_types().values():
+        if getattr(tab_type, "is_persistent", False):
+            tab_enabled = tab_type.is_enabled(course_module, settings, user=request.user)
+            update_tab(course_tabs, tab_type, tab_enabled)
 
-    return tabs_changed
+    # Save the tabs into the course if they have been changed
+    if not course_tabs == course_module.tabs:
+        course_module.tabs = course_tabs
+
+
+def _get_tab_panel_for_type(tab_type):
+    """
+    Returns a tab panel representation for the specified tab type.
+    """
+    tab_panel = EXTRA_TAB_PANELS.get(tab_type)
+    if tab_panel:
+        return tab_panel
+    return {
+        "name": tab_type.title,
+        "type": tab_type.name
+    }
```

Comment:
This is kind of a weird case because these dictionaries get turned into `CourseTab`s, and using  namedtuple means updating a lot of `CourseTab`'s methods to expect the namedtuple interface instead. And that means updating everything that uses `CourseTab`, etc. That's a more far-reaching change than the scope of this PR, I think.